core/src/components/button/button.ionic.scss

```
@@ -102,44 +101,24 @@
 // Button Sizes
 // -------------------------------------------------------------------------------
 
-/* Extra Small Button */
-:host(.button-xsmall) {
-  --padding-top: #{globals.$ion-space-100};
-  --padding-end: #{globals.$ion-space-300};
-
-  min-height: globals.$ion-space-600;
-
-  font-size: globals.$ion-font-size-300;
-}
-
 /* Small Button */
 :host(.button-small) {
   --padding-top: #{globals.$ion-space-200};
   --padding-end: #{globals.$ion-space-400};
 
-  min-height: #{globals.$ion-scale-800};
+  @include globals.typography(globals.$ion-body-action-sm);
 
-  font-size: #{globals.$ion-font-size-300};
+  min-height: #{globals.$ion-scale-1000};
 }
 
 /* Large Button */
 :host(.button-large) {
   --padding-top: #{globals.$ion-space-400};
   --padding-end: #{globals.$ion-space-600};
```

```suggestion
  --padding-end: #{globals.$ion-space-700};
```

`--padding-top` can be removed because it should be `0` which is defined in `:host`